src/analog/private/m2kanalogout_impl.cpp

```diff
@@ -449,6 +449,11 @@ class M2kAnalogOut::M2kAnalogOutImpl : public DeviceGeneric {
 		return m_dac_devices.at(chnIdx)->isChannelEnabled(0);
 	}
 
+	std::vector<double> getAvailableSampleRates(unsigned int chnIdx)
+	{
+		return m_dac_devices.at(chnIdx)->getAvailableSampleRates();
```

Comment:
curios: what happens at `m_dac_devices.at(chnIdx)->getAvailableSampleRates()`  when `m_dac_devices.at(chnIdx)` has no element?

Reply:
That's a good point. I added a getter that checks whether the index is valid or not.